### PR TITLE
Adding doc section for Mongo pumps indexes behavior

### PR DESCRIPTION
**Pull request by tbuchaillot**

### CODE REVIEW

**Review by marksou — APPROVED** [review]
LGTM

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md [hunk 1]
```diff
@@ -112,6 +112,15 @@ Let's discuss these pumps, their configs, matching collections and relevant dash
 ### Capping
 This collection [should be capped](/docs/tyk-configuration-reference/tyk-pump-configuration/tyk-pump-configuration/#capping-analytics-data) due to the number of individual documents. This is especially important if the `detailed_recording` in the Gateway is turned on which means that the Gateway records the full payload of the request and response. 
 
+### Omitting indexes
+From Pump 1.5.2+, the Mongo Pumps indexes default behaviour is changed and the new configuration option `omit_index_creation` is available. This apply for `Mongo Pump`,`Mongo Aggregate Pump` and `Mongo Selective Pump`.
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: From Pump 1.5.2+, a new configuration option `omit_index_creation` is available to allow control over the Pump's behavious with indexes.

This option (which can be set to 'true' or 'false') is applicable to the following Pumps: `Mongo Pump`,`Mongo Aggregate Pump` and `Mongo Selective Pump`; for any other Pump, the option does not exist.

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md [hunk 2]
```diff
@@ -112,6 +112,15 @@ Let's discuss these pumps, their configs, matching collections and relevant dash
...
+
+The behaviour now is:
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: The behaviour now depends upon the value of 'omit_index_creation' and the Pump in use, as follows:

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md [hunk 3]
```diff
@@ -112,6 +112,15 @@ Let's discuss these pumps, their configs, matching collections and relevant dash
...
+- If `omit_index_creation` is set to `true`, tyk-pump is not going to create any index for mongo pumps.
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: If `omit_index_creation` is set to `true`, tyk-pump will not create any indexes (for Mongo pumps).

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md [hunk 4]
```diff
@@ -112,6 +112,15 @@ Let's discuss these pumps, their configs, matching collections and relevant dash
...
+- If `omit_index_creation` is set to `false` (default) and you are using `DocumentDB`, tyk-pump is going to create the mongo indexes.
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: If `omit_index_creation` is set to `false` (default) and you are using `DocumentDB`, tyk-pump will create the Mongo indexes.

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md [hunk 5]
```diff
@@ -112,6 +112,15 @@ Let's discuss these pumps, their configs, matching collections and relevant dash
...
+- If `omit_index_creation` is set to `false` (default) and you are using `MongoDB`, we have 2 behaviors:
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: If 'omit_index_creation' is set to 'false' (default) and you are using 'MongoDB', the behaviour of tyk-pump depends upon whether the collection already exists:

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md [hunk 6]
```diff
@@ -112,6 +112,15 @@ Let's discuss these pumps, their configs, matching collections and relevant dash
...
+  - If the collection exists, tyk-pump is not going to create the indexes again.
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: If the collection already exists, tyk-pump will not create the indexes again.

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md [hunk 7]
```diff
@@ -112,6 +112,15 @@ Let's discuss these pumps, their configs, matching collections and relevant dash
...
+  - If the collection doesn't exist, tyk-pump is going to create the indexes.
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: If the collection does not already exist, tyk-pump will create the indexes.

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md (hunk 1 above)
Comment (andyo-tyk) [reply tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: Question: for non-Mongo pumps, does the option simply not exist (i.e. will throw an error if set) or is it simply ignored?

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Proposed new wording; one question on behaviour for non-Mongo Pumps

**Review by sredxny — APPROVED** [review]
lgtm

File: tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md (hunk 1 above)
Comment (tbuchaillot) [reply tyk-docs/content/tyk-stack/tyk-pump/tyk-pump-configuration/tyk-pump-dashboard-config.md]: @andyo-tyk that option doesn't exist for other pumps, not every data store uses indexes nor has configurations about it. It's just for mongo pumps.

**Comment by tbuchaillot** [comment]
@andyo-tyk please have another look 👍

**Review by andyo-tyk — APPROVED** [review]
All good now, thanks.

**Comment by tbuchaillot** [comment]
@marksou can we merge it?

**Pull request merged**